src/ToolBox/superpmi/superpmi/neardiffer.cpp

```diff
@@ -67,14 +67,51 @@ bool NearDiffer::InitAsmDiff()
 
     if (UseCoreDisTools)
     {
-        const char* coreDisToolsLibrary = MAKEDLLNAME_A("coredistools");
+#ifdef PLATFORM_UNIX
+        // Unix will require the full path to coredistools. Assume that the
+        // location is next to the full path to the superpmi.so.
+        
+        wchar_t coreCLRLoadedPath[MAX_LONGPATH];
+        HMODULE result = 0;
+        int returnVal = ::GetModuleFileNameW(result, coreCLRLoadedPath, MAX_LONGPATH);
+
+        if (returnVal == 0)
+        {
+            LogError("GetModuleFileNameW failed (0x%08x)", ::GetLastError());
+            return false;
+        }
+
+        wchar_t* ptr;
+        for (ptr = coreCLRLoadedPath + ::wcsnlen(coreCLRLoadedPath, MAX_LONGPATH) - 1; ptr != coreCLRLoadedPath; --ptr)
+        {
+            if (*ptr == '/')
+            {
+                ++ptr;
+                break;
+            }
+        }
+
+        const wchar_t* coreDisToolsLibrary = MAKEDLLNAME_W("coredistools");
+        ::wcscpy_s(ptr, ::wcsnlen(coreDisToolsLibrary, MAX_LONGPATH) + 1, coreDisToolsLibrary);
 
+        HMODULE hCoreDisToolsLib = ::LoadLibraryW(coreCLRLoadedPath);
+        if (hCoreDisToolsLib == 0)
+        {
+            LogError("LoadLibrary(%s) failed (0x%08x)", ::GetLastError());
+            return false;
+        }
+
+#else
+        const char* coreDisToolsLibrary = MAKEDLLNAME_A("coredistools");
         HMODULE hCoreDisToolsLib = ::LoadLibraryA(coreDisToolsLibrary);
```

Comment:
Is not is possible to use the wide char encoding for Unix/Windows here and merge the tail:
```
        const wchar_t* coreDisToolsLibrary = MAKEDLLNAME_W("coredistools");
#ifdef PLATFORM_UNIX
#endif // PLATFORM_UNIX
        HMODULE hCoreDisToolsLib = ::LoadLibraryW(coreCLRLoadedPath);
        if (hCoreDisToolsLib == 0)
        {
            LogError("LoadLibrary(%s) failed (0x%08x)", ::GetLastError());
            return false;
        }
```

Reply:
The code to do so is significantly more complex, because you then need to look for special backtick characters. Plus, the OS will do all of this work for us on Windows, so it ends up being a waste of cycles to do it here.

Reply:
Nvm I understand further what you are saying. Will change.